pds_doi_core/db/doi_database.py

```diff
@@ -350,6 +351,70 @@ def select_row_one(self, db_name, table_name, query_criterias):
 
         return 1
 
+    def create_q_string_for_latest_rows(self, table_name, query_criterias):
+        ''' Build the query string to select all rows with column is_latest = 1 in ascending order.'''
+
+        query_string = 'SELECT * FROM ' + table_name
+
+        # If there are some criteria provided by user, we start with the ' WHERE ' clause.
+        if len(query_criterias) > 0:
+            query_string += ' WHERE '
+        for ii in range(len(query_criterias)):
+            # Build the WHERE clause
+            if ii == 0:
+                query_string += query_criterias[ii]
+            else:
+                query_string += ' AND '   + query_criterias[ii]
```

Comment:
It is better to use the ? syntax to have criteria in a SQL request. Regarding security is prevent SQL injection (see https://www.w3schools.com/sql/sql_injection.asp). You have done that for other request, it is good to keep this practice.

Reply:
After looking over the actual query, the criterias is not needed so it has been removed.

Reply:
From the comment I made after, I think you will need to re-introduce the criteria as the list command needs these criteria. This will requires some thinking on how generate the prepared statement (the string request with ?s) for a variable list of criteria (all of them are optional).

Reply:
The criteria is now a dict object.  The db object will convert the query string either with '=' for regular fields or '<=' and '>=' for time related WHERE clause.  The question mark '?' will be used to avoid sql-injection with the binding taking place with the execute() function is called.

Will get the criteria from user to db once we get your latest changes for subparser into my branch #32.